lib/pfconfig/namespaces/resource/passthroughs.pm

```diff
@@ -0,0 +1,113 @@
+package pfconfig::namespaces::resource::passthroughs;
+
+=head1 NAME
+
+pfconfig::namespaces::resource::passthroughs
+
+=cut
+
+=head1 DESCRIPTION
+
+pfconfig::namespaces::resource::passthroughs
+
+=cut
+
+use strict;
+use warnings;
+
+use base 'pfconfig::namespaces::resource';
+
+sub init {
+    my ($self) = @_;
+    $self->{config}         = $self->{cache}->get_cache('config::Pf');
+    $self->{authentication_sources} = $self->{cache}->get_cache('resource::authentication_sources');
+}
+
+sub build {
+    my ($self) = @_;
+
+    my @all_passthroughs = (
+        @{$self->{config}->{trapping}->{passthroughs}}, 
+        map{
+            $_->isa("pf::Authentication::Source::OAuthSource") 
+                ? split(/\s*,\s*/, $_->{domains})
+                : () 
+        } @{$self->{authentication_sources}},
+    );
+
+    my %passthroughs = (
+        normal => {}, 
+        wildcard => {},  
+    );
+    foreach my $passthrough (@all_passthroughs) {
+        my ($domain, $ports) = $self->_new_passthrough($passthrough);
+        my $ns = "normal";
+        if($domain =~ /\*\.(.*)/) {
+            $ns = "wildcard";
+            $domain = $1;
+        }
+        if(defined($passthroughs{$ns}{$domain})) {
+            push @{$passthroughs{$ns}{$domain}}, @$ports;
+        }
+        else {
+            $passthroughs{$ns}{$domain} = $ports;
+        }
+    }
+
+    return \%passthroughs;
+}
+
+sub _new_passthrough {
+    my ($self, $passthrough) = @_;
+
+    if($passthrough =~ /(.*?):(udp:|tcp:)?([0-9]+)/) {
+        my $domain = $1;
+        # NOTE: proto contains the ':' at the end
+        my $proto = $2;
+        my $port = $3;
+        if($proto) {
+            return ($domain, [$proto.$port]);
+        }
+        else {
+            return ($domain, ["udp:$port", "tcp:$port"]);
```

Comment:
If I add a port with no proto there three different way of interrupting it.

* Only tcp
* Only udp
* Both

Some people would think it will only open tcp since you only added a single port like I did.
Other would expect both
Almost none would expect udp 

In either case it is not properly documented behavior.